src/FffGcodeWriter.cpp

```diff
@@ -868,9 +876,17 @@ void FffGcodeWriter::addMeshPartToGCode(SliceDataStorage& storage, SliceMeshStor
         mesh->getSettingAsFillMethod("top_bottom_pattern_0") :
         mesh->getSettingAsFillMethod("top_bottom_pattern");
     int skin_angle = 45;
-    if ((skin_pattern == EFillMethod::LINES || skin_pattern == EFillMethod::ZIG_ZAG) && layer_nr & 1)
+    if (skin_pattern == EFillMethod::LINES || skin_pattern == EFillMethod::ZIG_ZAG)
     {
-        skin_angle += 90; // should coincide with infill_angle (if both skin and infill are lines) so that the first top layer is orthogonal to the last infill layer
+        if (mesh->infill_angles.size() > 0)
```

Comment:
`infill_angles` cannot be empty if you fixed my above comment.

Please make sure the infill angles stay the same for all the infill types (except the concentric ones)